content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,229 @@
+---
+title: "Design Istio Ambient to Work with Any Platforms and CNIs"
+description: An innovative traffic redirection mechanism between workload pods and ztunnel.
+publishdate: 2024-01-26
+attribution: "Ben Leggett (Solo.io), Yuval Kohavi (Solo.io), Lin Sun (Solo.io)"
+keywords: [Ambient,Istio,CNI,ztunnel,traffic]
+---
+
+## Introduction
+
+Since the initial [launch](/blog/2022/introducing-ambient-mesh/) of ambient in 2022 and ambient reaching
+[Alpha](/news/releases/1.18.x/announcing-1.18/#ambient-mesh) status in early 2023 , the Istio community has
+been focused on stabilizing Ambient for beta, where we can start recommending users to run Ambient in production
+with caution. Through various GitHub issues and slack messages, we heard strong desires to support ambient
+for [minikube](https://github.com/istio/istio/issues/46163), [Cilium CNI](https://github.com/istio/istio/issues/44198),
+[Calico CNI](https://github.com/istio/istio/issues/40973), [Docker Desktop for Kubernetes](https://github.com/istio/istio/issues/47436), [OpenShift](https://github.com/istio/istio/issues/42341),
+[AWS Cloud](https://github.com/istio/istio/issues/42340) etc. This has become the No. 1 requirement for Ambient
+Beta - support Ambient to work with any Platforms or CNIs, maintaining Istio’s existing broad and general compatibility
+with various Kubernetes platforms with different Container Network Interface(CNI) layers. After all,
+Ambient wouldn’t be ambient without being able to run everywhere with any CNIs.
+
+At Solo, we had been working hard for 9 months on an innovative internal solution to solve this and we had decided
+to [upstream](https://github.com/istio/istio/issues/48212) our changes in late 2023 to help Ambient reach beta faster
+so more users can operate ambient in Istio 1.21 or newer and enjoy the benefits of ambient sidecar-less mesh in their
+platforms regardless of their preferred CNIs.
+
+## How do we get here?
+
+### Istio and CNI - What’s The Relationship?
+
+Istio is a service mesh, and service meshes by definition are not *CNI implementations* - service meshes build on top of the
+low-level, standards-compliant CNI implementation present in Kubernetes clusters to offer the high-level policy and security
+controls users need at scale.
+
+Basically, before you can do things like secure pod traffic with mTLS and apply lots of high-level authentication and authorization policy
+at the service mesh layer, you must have a functional Kubernetes cluster with a functional CNI implementation, to make sure the
+basic networking pathways are set up so that packets can get from one pod to another (and from one node to another) in
+your cluster.
+
+Though some service meshes may also ship and require their own in-house CNI implementation, the Istio project has chosen not
+to, for maximum compatibility with managed offerings, cross-vendor support, and composability with the broader CNCF ecosystem.
```

Comment:
again, address the fact that Istio has a CNI, and possibly introduce what the difference between [an interface plugin and a chained plugin](https://www.cni.dev/docs/spec/#overview-1)

Reply:
Istio doesn't have a CNI implementation (which is a large and complex thing), it extends whatever the CNI implementation you are using (Calico, Cilum, cloud provider variant) with a CNI plugin.

Again, will reword. It's a bit messy. We call the Istio component a "CNI agent" but it does not provide a CNI implementation and in fact requires one to already be in place.

Reply:
"istio-cni" being a CNI itself is a common confusion. I recommend tackling this head-on with what you said.